### PR TITLE
#12 Inital plumbing / architecture

### PR DESCRIPTION
**Pull request by ebarriosara**
Same configuration as in https://github.com/Lemoncode/react-simple-testing/tree/master/00%20Boilerplate has been done, except for the testing configuration, that has been followed from https://github.com/Lemoncode/redux-by-sample/tree/master/11%20Testing_Jest because the first one was using karma, mocha, sinon, etc. instead of just jest and enzyme like the second one.

Apart from that, there are some other dependencies that have not been added because it is not sure that we will need them. They are: bootstrap, jquery, lc-form-validation and toastr.

Regarding webpack files, common.js has not been created because we won't have a test folder for webpack. Instead, we will only have one webpack folder with base.js, dev.js and prod.js.

Please, let me know if something else should be changed, added or removed. Thanks.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+node_modules
+.DS_Store
+dist/
+*.orig
+.idea/
+bundle.js
+package-lock.json
```
Comment (arp82) [inline .gitignore:7]: Just to make sure, have we agreed on removing "package-lock.json" from the set of tracked files? I am not against the idea, but I wanted to double-check (as its inclusion as part of the source files is usually recommended by the NPM team)

Comment (brauliodiez) [reply .gitignore:7]: We have had issues with package.lock.json (it gets corrupted from time to time :-( ), we should some day give a try though

**Comment by brauliodiez** [comment]
Good stuff, next step it to add routing, here we will add version 3 of react-router (@Nasdan can provide you more detail about this), version 4 still doesn't integrate seamlessly with redux in case we need it.

File: config/webpack/base.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  },
+  plugins: [
+    new webpack.optimize.CommonsChunkPlugin({
+      names: ['vendor', 'manifest'],
+    }),
+    new HtmlWebpackPlugin({
+      filename: 'index.html',
+      template: 'index.html',
+      hash: true,
+      chunks: ['manifest', 'vendor', 'vendorStyles', 'appStyles', 'app'],
```
Comment (nasdan) [inline config/webpack/base.js:89]: Remove vendorStyles, you have no entry bundle

File: config/helpers.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const path = require('path');
+
+const rootPath = path.resolve(__dirname, '..');
+
+exports.resolveFromRootPath = (...args) => path.join(rootPath, ...args);
```
Comment (nasdan) [inline config/helpers.js:5]: When the last symbol appears, means that you need empty new line

File: config/webpack/dev.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  output: {
+    path: helpers.resolveFromRootPath('dist'),
+    filename: '[name].js',
+  },
+  devServer: {
+    contentBase: helpers.resolveFromRootPath('dist'),
+    inline: true,
+    host: 'localhost',
+    port: 8080,
+    stats: 'errors-only',
```
Comment (nasdan) [inline config/webpack/dev.js:28]: Replace by `minimal` because with errors-only we don't see tslint warnings

File: package.json [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    "tslint": "^5.9.1",
+    "tslint-react": "^3.5.1",
+    "typescript": "^2.7.2",
+    "url-loader": "^0.6.2",
+    "webpack": "^3.11.0",
+    "webpack-dev-server": "^2.11.1",
+    "webpack-merge": "^4.1.1"
+  },
+  "jest": {
+    "testRegex": "\\.spec\\.tsx?$",
```
Comment (nasdan) [inline package.json:78]: Pending to add jest configuration, did you read all Jest sample? During the specs we continue with configuration:

````
"testRegex": "(/specs/.*|\\.spec)\\.(ts|tsx)$",
    "moduleFileExtensions": [
      "js",
      "jsx",
      "json",
      "ts",
      "tsx"
    ],
    "transform": {
      ".tsx?": "<rootDir>/node_modules/ts-jest/preprocessor.js"
    },
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
    "moduleNameMapper": {
      "^.+\\.s?css$": "identity-obj-proxy"
    },
    "setupFiles": [
      "<rootDir>/config/test/polyfills.js",
      "<rootDir>/config/test/setupTest.js"
    ],
    "restoreMocks": true
```

File: .env [hunk 6]
```diff
@@ -0,0 +1,2 @@
+REST_ENV=mock
```
Comment (nasdan) [inline .env:1]: You should ignore this file and does not upload to repository, it's an environment that we would like to replace in production, development, etc.

**Comment by ebarriosara** [comment]
Thanks for the review.

All issues, except for the .env one, have been committed in #17.

Regarding .env file, we need to clarify what to do, because we are using those variables in start:prod and build:prod scripts in package.json.

**Pull request merged**